### DISCUSSION TITLE
Strapi field type compatible with ObjectId field type in MongoDB

### DISCUSSION THREAD

**Discussion by khanhtmn** [strapi/strapi#7545] [Legacy Topics]
Hi I'm trying to plug in my MongoDB database to the backend of Strapi and having difficulty in field validation. I have a field `productId` which has the type of `ObjectId` and I don't know which field type in Strapi is compatible with `ObjectId`.

I tried `text` , `richtext`, `UID`, but all of them doesn't enable filter. For example, if I have a `productId = 12345` in the database, I cannot use Strapi filter to fetch the record which has `productId = 12345` . It always shown as "No record found" .

Any leads on this?

**Comment by derrickmehaffy** [comment]
Hello!

As I replied to you in slack, it is currently not recommended to attach a database to Strapi that wasn't created by Strapi, especially with MongoDB. In this case if productId is your primary key it will not work with Strapi as currently you can't customize the name of the primary key: https://strapi.io/documentation/v3.x/getting-started/troubleshooting.html#does-strapi-allow-me-to-change-the-default-id-type-or-name

With Strapi and MongoDB, currently we don't use Mongo "as it was intended" with child documents and more or less treat MongoDB as a relational database, even though it is not.

Can you please clarify your use-case here?

**Reply by khanhtmn** [reply]
Hi @derrickmehaffy ,

Many thanks for getting back to me so quickly. I looked at the documentation you shared and have a question: 
> ....it doesn't allow you to switch the data type (such as UUID on bookshelf and integer on mongoose)

 Does it mean that if we have a field type `integer` on the backend/mongoose, we cannot declare it as `UUID` on Strapi? Then what would be a Strapi field type that is compatible with `ObjectId` field type on MongoDB?

I'm more than happy to clarify my use case here - glad you asked! I have 2 tables: 
- The first table is `product` which has field `id` as primary key.  `id` has a type of `ObjectId` because this is by default of MongoDB. However, it is filterable by Strapi filter.
- The second table is `order` which reference to the products from the table `product` as field `productId`. `productId`  also has a type of `ObjectId` but it is not filterable by Strapi filter. 

So my question is: How to make `productId` filterable? The filter feature is the sole reason why I adopted Strapi in the first place TT^TT

Thank you in advance!

**Reply by derrickmehaffy** [reply]
In your case the `productId` would be a relation and doesn't really have "type" defined in Strapi, but most likely it would be just a string (text). As we don't have a specific type for ObjectIDs.
